@xen-orchestra/web/src/types/xo/new-vm.type.ts

```diff
@@ -0,0 +1,46 @@
+import type { XoHost } from '@/types/xo/host.type.ts'
+import type { XoPool } from '@/types/xo/pool.type'
+import type { XoSr } from '@/types/xo/sr.type.ts'
+import type { XoVdi } from '@/types/xo/vdi.type.ts'
+import type { XoVmTemplate } from '@/types/xo/vm-template.type'
+
+export interface Vdi {
+  name_label: string
+  name_description: string
+  size: number
+  sr: XoSr['id'] | undefined
+}
+
+export interface NetworkInterface {
+  interface: string
+  macAddress: string
+}
+
+export interface VmState {
+  affinity_host: XoHost['id'] | undefined
+  auto_poweron: boolean
+  installMode: string
+  boot_firmware: string
+  boot_vm: boolean
+  clone: boolean
+  cloudConfig: string
+  copyHostBiosStrings: boolean
+  defaultNetwork: NetworkInterface | undefined
+  isDiskTemplateSelected: boolean
+  networkConfig: string
+  networkInterfaces: NetworkInterface[]
+  new_vm_template: XoVmTemplate | undefined
+  pool: XoPool | undefined
+  ram: number
+  selectedVdi: XoVdi['id'] | undefined
+  sshKeys: string[]
+  ssh_key: string
+  tags: string[]
+  topology: string
+  vCPU: number
+  selectedVCPU: number
+  vdis: string[]
+  description: string
+  existingVdis: Vdi[]
```

Comment:
Many types are marked as non-optional, but potentially undefined. Is this the desired behavior?
`foo?: string` and `foo: string | undefined` are very different.
The first means that `foo` is optional (undefined) or a string, while the other: `foo` is not optional, but can be undefined or a string.

E.g.
```ts
function getFoo(foo?: string){
 return foo ?? "bar"
}
getFoo() // OK
getFoo("foo") // OK
getFoo(undefined) // OK

function getBar(bar: string | undefined){
 return bar ?? "bar"
}
getBar() // KO
getBar("bar") // OK
getBar(undefined) // OK
```